lib/matplotlib/ticker.py

```diff
@@ -1120,15 +1114,13 @@ def __call__(self, x, pos=None):
                     '{0}{1:g}'.format(sign_string, x)))
         elif not is_x_decade:
             return self._non_decade_format(sign_string, base, fx, usetex)
+        elif usetex:
+            return r'$%s%s^{%d}$' % (
+                sign_string, base, int(fx + (-.5 if fx < 0 else .5)))
```

Comment:
actually fx is close to int here (the "not is_x_decade" case is handled just above) so it has already been rounded above, so you don't need to call any of nearest_long, round, or your inlined version.